src/analysis/complex/exponential.lean

```diff
@@ -1885,3 +1900,111 @@ lemma has_deriv_within_at.rexp {f : ℝ → ℝ} {f' x : ℝ} {s : set ℝ}
   (hf : has_deriv_within_at f f' s x) :
   has_deriv_within_at (real.exp ∘ f) (f' * real.exp (f x)) s x :=
 (real.has_deriv_at_exp (f x)).comp_has_deriv_within_at x hf
+
+namespace nnreal
+
+/-- The nonnegative real power function `x^y`, defined for `x : nnreal` and `y : ℝ ` as the
+restriction of the real power function. For `x > 0`, it is equal to `exp (y log x)`. For `x = 0`,
+one sets `0 ^ 0 = 1` and `0 ^ y = 0` for `y ≠ 0`. -/
+noncomputable def rpow (x : nnreal) (y : ℝ) : nnreal :=
+⟨(x : ℝ) ^ y, real.rpow_nonneg_of_nonneg x.2 y⟩
+
+noncomputable instance : has_pow nnreal ℝ := ⟨rpow⟩
+
+@[simp, push_cast] lemma coe_rpow (x : nnreal) (y : ℝ) : ((x ^ y : nnreal) : ℝ) = (x : ℝ) ^ y := rfl
```

Comment:
I believe it is called `move_cast` (at least until #1103 will rename all these attrs to `norm_cast`).

Reply:
Done